magic.js

```diff
@@ -1009,6 +1018,71 @@ function dsync(sk, ciphertext, nonce, cb) {
   return done(null, convert(plaintext));
 }
 
+module.exports.pwdDecrypt.aead = (p, c, n, cb) => {
+  return sodium.ready.then(() => {
+    if (typeof p === 'function') {
+      cb = p;
+      p = null;
+    }
+    const done = ret(cb);
+
+    if (!p) { return done(new Error('Cannot decrypt without a password')); }
+
+    let s;
+    [ c ] = cparse(c);
+    salt = c.slice(0, sodium.crypto_pwhash_SALTBYTES);
+    try {
+      s = sodium.crypto_pwhash(
+        KEY_SIZE,
+        p,
+        salt,
+        sodium.crypto_pwhash_OPSLIMIT_INTERACTIVE,
+        sodium.crypto_pwhash_MEMLIMIT_INTERACTIVE,
+        sodium.crypto_pwhash_ALG_DEFAULT
+      );
+    } catch(ex) {
+      return done(new Error('Libsodium error: ' +  ex))
```

Comment:
qq: do you know how we can trigger a libsodium related error?
I'd like to add at least one test case that hits this line for expected error messages to prevent any type of info leak from libsodium since we're currently relying on them to sanitize their error messages.